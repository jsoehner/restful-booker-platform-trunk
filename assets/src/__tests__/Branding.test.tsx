import React from 'react';
import BrandingForm from '../components/admin/Branding';
import ReactModal from 'react-modal';
import { render, fireEvent, waitFor } from '@testing-library/react';

const brandingData = {
  name: 'Shady Meadows B&B',
  map: {
    latitude: 52.6351204,
    longitude: 1.2733774
  },
  directions: 'Take the first left after the big tree, then follow the road until you see the sign.',
  logoUrl: 'https://www.mwtestconsultancy.co.uk/img/rbp-logo.png',
  description: 'Welcome to Shady Meadows, a delightful Bed & Breakfast nestled in the hills on Newingtonfordburyshire. A place so beautiful you will never want to leave. All our rooms have comfortable beds and we provide breakfast from the locally sourced supermarket. It is a delightful place.',
  contact: {
    name: 'Shady Meadows B&B',
    phone: '0123456789',
    email: 'fake@fakeemail.com'
  },
  address: {
    line1: '123 Fake Street',
    line2: 'Fake Town',
    postTown: 'Fake Town',
    county: 'Fake County',
    postCode: 'FA1 2KE'
  }
};

describe('Branding Component', () => {
  beforeEach(() => {
    jest.clearAllMocks();
    global.fetch = jest.fn()
      .mockImplementationOnce(() => Promise.resolve({
        ok: true,
        json: () => Promise.resolve(brandingData)
      }));
  });

  test('Branding page renders', async () => {
    const { asFragment, findByDisplayValue } = render(
      <BrandingForm />
    );

    await findByDisplayValue("52.6351204");
    
    expect(asFragment()).toMatchSnapshot();
  });

  test('Branding page shows modal on success', async () => {
    global.fetch = jest.fn()
      .mockImplementationOnce(() => Promise.resolve({
        ok: true,
        json: () => Promise.resolve(brandingData)
      }))
      .mockImplementationOnce(() => Promise.resolve({
        ok: true
      }));

    ReactModal.setAppElement(document.createElement('div'));

    const { getByText, getByPlaceholderText } = render(
      <BrandingForm />
    );

    fireEvent.change(getByPlaceholderText('Enter B&B name'), { target: { value: 'Updated Room' } });
    fireEvent.click(getByText('Submit'));

    await waitFor(() => expect(getByText('Branding updated!')).toBeInTheDocument());
  });

  test('Branding page shows errors', async () => {
    global.fetch = jest.fn()
      .mockImplementationOnce(() => Promise.resolve({
        ok: true,
        json: () => Promise.resolve(brandingData)
      }))
      .mockImplementationOnce(() => Promise.resolve({
        ok: false,
        json: () => Promise.resolve({
          fieldErrors: ["Phone should not be blank"]
        })
      }));

    const { getByText, findByText } = render(
      <BrandingForm />
    );

    fireEvent.click(getByText('Submit'));
    await findByText('Phone should not be blank');

    expect(getByText('Phone should not be blank')).toBeInTheDocument();
  });
});
