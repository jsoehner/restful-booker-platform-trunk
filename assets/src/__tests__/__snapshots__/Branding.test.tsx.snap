// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Branding Component Branding page renders 1`] = `
<DocumentFragment>
  <div
    class="branding-form"
  >
    <h2>
      B&B details
    </h2>
    <div
      class="input-group mb-3"
    >
      <div
        class="input-group-prepend d-flex align-items-center"
      >
        <span
          class="input-group-text"
        >
          Name
        </span>
      </div>
      <input
        class="form-control"
        id="name"
        placeholder="Enter B&B name"
        type="text"
        value="Shady Meadows B&B"
      />
    </div>
    <div
      class="input-group mb-3"
    >
      <div
        class="input-group-prepend d-flex align-items-center"
      >
        <span
          class="input-group-text"
        >
          Logo
        </span>
      </div>
      <input
        class="form-control"
        id="logoUrl"
        placeholder="Enter image url"
        type="text"
        value="https://www.mwtestconsultancy.co.uk/img/rbp-logo.png"
      />
    </div>
    <div
      class="input-group mb-3"
    >
      <div
        class="input-group-prepend d-flex align-items-stretch"
      >
        <span
          class="input-group-text"
        >
          Description
        </span>
      </div>
      <textarea
        class="form-control"
        id="description"
        rows="5"
      >
        Welcome to Shady Meadows, a delightful Bed & Breakfast nestled in the hills on Newingtonfordburyshire. A place so beautiful you will never want to leave. All our rooms have comfortable beds and we provide breakfast from the locally sourced supermarket. It is a delightful place.
      </textarea>
    </div>
    <br />
    <h2>
      Map details
    </h2>
    <div
      class="input-group mb-3"
    >
      <div
        class="input-group-prepend d-flex align-items-center"
      >
        <span
          class="input-group-text"
        >
          Latitude
        </span>
      </div>
      <input
        class="form-control"
        id="latitude"
        placeholder="Enter Latitude"
        type="text"
        value="52.6351204"
      />
    </div>
    <div
      class="input-group mb-3"
    >
      <div
        class="input-group-prepend d-flex align-items-center"
      >
        <span
          class="input-group-text"
        >
          Longitude
        </span>
      </div>
      <input
        class="form-control"
        id="longitude"
        placeholder="Enter Longitude"
        type="text"
        value="1.2733774"
      />
    </div>
    <div
      class="input-group mb-3"
    >
      <div
        class="input-group-prepend d-flex align-items-stretch"
      >
        <span
          class="input-group-text"
        >
          Directions
        </span>
      </div>
      <textarea
        class="form-control"
        id="directions"
        rows="5"
      >
        Take the first left after the big tree, then follow the road until you see the sign.
      </textarea>
    </div>
    <br />
    <h2>
      Contact details
    </h2>
    <div
      class="input-group mb-3"
    >
      <div
        class="input-group-prepend d-flex align-items-center"
      >
        <span
          class="input-group-text"
        >
          Name
        </span>
      </div>
      <input
        class="form-control"
        id="contactName"
        placeholder="Enter Contact Name"
        type="text"
        value="Shady Meadows B&B"
      />
    </div>
    <div
      class="input-group mb-3"
    >
      <div
        class="input-group-prepend d-flex align-items-center"
      >
        <span
          class="input-group-text"
        >
          Phone
        </span>
      </div>
      <input
        class="form-control"
        id="contactPhone"
        placeholder="Enter Phone Number"
        type="text"
        value="0123456789"
      />
    </div>
    <div
      class="input-group mb-3"
    >
      <div
        class="input-group-prepend d-flex align-items-center"
      >
        <span
          class="input-group-text"
        >
          Email
        </span>
      </div>
      <input
        class="form-control"
        id="contactEmail"
        placeholder="Enter Email Address"
        type="email"
        value="fake@fakeemail.com"
      />
    </div>
    <br />
    <h2>
      Address details
    </h2>
    <div
      class="input-group mb-3"
    >
      <div
        class="input-group-prepend d-flex align-items-center"
      >
        <span
          class="input-group-text"
        >
          Line 1
        </span>
      </div>
      <input
        class="form-control"
        id="line1"
        placeholder="Enter Address Line 1"
        type="text"
        value="123 Fake Street"
      />
    </div>
    <div
      class="input-group mb-3"
    >
      <div
        class="input-group-prepend d-flex align-items-center"
      >
        <span
          class="input-group-text"
        >
          Line 2
        </span>
      </div>
      <input
        class="form-control"
        id="line2"
        placeholder="Enter Address Line 2"
        type="text"
        value="Fake Town"
      />
    </div>
    <div
      class="input-group mb-3"
    >
      <div
        class="input-group-prepend d-flex align-items-center"
      >
        <span
          class="input-group-text"
        >
          Post Town
        </span>
      </div>
      <input
        class="form-control"
        id="postTown"
        placeholder="Enter Post Town"
        type="text"
        value="Fake Town"
      />
    </div>
    <div
      class="input-group mb-3"
    >
      <div
        class="input-group-prepend d-flex align-items-center"
      >
        <span
          class="input-group-text"
        >
          County
        </span>
      </div>
      <input
        class="form-control"
        id="county"
        placeholder="Enter County"
        type="text"
        value="Fake County"
      />
    </div>
    <div
      class="input-group mb-3"
    >
      <div
        class="input-group-prepend d-flex align-items-center"
      >
        <span
          class="input-group-text"
        >
          Post Code
        </span>
      </div>
      <input
        class="form-control"
        id="postCode"
        placeholder="Enter Post Code"
        type="text"
        value="FA1 2KE"
      />
    </div>
    <br />
    <button
      class="btn btn-outline-primary"
      id="updateBranding"
      type="submit"
    >
      Submit
    </button>
  </div>
</DocumentFragment>
`;
